### PR TITLE
chore: move common dependencies to workspace Cargo.toml

### PR DESCRIPTION
**Pull request by karimElmougi**
It's easier to keep versions in sync across the entire workspace this way.

### CODE REVIEW

**Comment by sxyazi** [comment]
Thanks for the PR! 

Every crate of Yazi gets published to crates.io, will this have any impact on it?

**Comment by karimElmougi** [comment]
I don't believe it should cause any issues. [quickwit](https://github.com/quickwit-oss/quickwit/tree/main/quickwit) makes use of workspace dependencies as well, and [their crates get published as well](https://crates.io/search?q=quickwit).

**Comment by sxyazi** [comment]
> I don't believe it should cause any issues. [quickwit](https://github.com/quickwit-oss/quickwit/tree/main/quickwit) makes use of workspace dependencies, and [their crates get published as well](https://crates.io/search?q=quickwit).

Nice, TIL about that.

File: yazi-adapter/Cargo.toml [hunk 1]
```diff
@@ -13,22 +13,22 @@ yazi-config = { path = "../yazi-config", version = "0.3.0" }
...
-arc-swap     = "1.7.1"
-base64       = "0.22.1"
+ansi-to-tui  = { workspace = true }
+anyhow       = { workspace = true }
+arc-swap     = { workspace = true }
+base64       = { workspace = true }
 color_quant  = "1.1.0"
-crossterm    = "0.27.0"
-futures      = "0.3.30"
+crossterm    = { workspace = true }
```
Comment (sxyazi) [inline yazi-adapter/Cargo.toml:21]: We can't use `{ workspace = true }` here because the "use-dev-tty" feature can only be enabled on macOS systems.

**Review by sxyazi — CHANGES_REQUESTED** [review]

File: yazi-adapter/Cargo.toml (hunk 1 above)
Comment (karimElmougi) [reply yazi-adapter/Cargo.toml:21]: Ah, I see. I hadn't realized. I pushed a fix.

**Review by sxyazi — APPROVED** [review]
LGTM, thanks!

**Pull request merged**